### ISSUE TITLE
Expand verify button to allow for manager review of CHW reports

### ISSUE THREAD

**Issue by sglangevin** [medic/medic-webapp#4529]
This feature adds a new option for reports so that in addition to marking a report verified, a manager can mark a report as having errors. This makes it easier for a manager to see which reports have errors and follow up with the CHWs who have submitted those reports.

Currently, clicking the verify button from the report detail page marks a report as verified. With this new feature, clicking the verify button shows a small menu with two options: Correct and Has errors. Adding this third state for "has errors" supports follow-up of CHWs as well as follow-up of patients who might have been affected by the errors.

Blocked on #4575.

Requirements:
- Report status (unreviewed, has errors, correct) must be shared with analytics and may need to be used for generating tasks or for certain targets

Interaction notes:
- Clicking on the checkmark icon opens or closes the review panel
- Only one status can be selected at a time
- Clicking on the currently selected status deselects that status

UI elements:
- [ ] new panel (56px tall) with the has errors and correct options in it; panel stretches the width of the screen on mobile and the width of the RHS action bar on desktop
- [ ] new panel has drop-shadow that matches what we are using when you click on the "new action" button on the people tab
- [ ] icons are 24x24px, 6px from the top edge of the panel; unselected icons are gray (#A7A9AC), has errors is red (#E0381F) when selected, correct is green (#789A11) when selected
- [ ] text is 14px, bold; when icon is unselected the text is gray (#777777), when icon is selected, the text color matches the icon color
- [ ] icons and text are centered within their half of the panel
- [ ] update the translation from Verify to Review

![image](https://user-images.githubusercontent.com/1286381/39693580-2b53736c-51ed-11e8-97ff-2ad049d8c560.png) ![image](https://user-images.githubusercontent.com/1286381/39693814-fb892414-51ed-11e8-9e39-82b1565c10f8.png) ![image](https://user-images.githubusercontent.com/1286381/39693837-09044920-51ee-11e8-8a8e-6d4f7e141446.png)

More information available [in this doc](https://docs.google.com/document/d/1lRoAVszPQVbhKFTp9ddYg0heVY39hsOXLr3P4t92_r4/edit#).

Acceptance criteria:
- clicking on the checkmark on the report detail page opens a panel with has errors and correct as options
- panel is correct size and layout
- clicking on one of the options selects it and changes the color
- clicking on the currently selected option deselects that option
- correct current status is stored in the db for use in analytics

**Comment by garethbowen** [comment]
@sglangevin Should the verify button only be shown to managers? Should it only be shown to the specific manager of the CHW who submitted this report?

**Comment by sglangevin** [comment]
The verify button would ideally only be shown to managers, but that is not a strict requirement of this feature. We current show the verify button for everyone and it isn't used. If it's easy to have a permission for showing/hiding the verify button, we can do that, but it's a nice-to-have and could be added later when we do more work on roles & permissions. @derickl we'll only need to define a manager if we choose to only show this button for managers, and in that case, we'd use roles/permissions.

**Comment by garethbowen** [comment]
We don't have a supervisor/manager role at the moment so we can't achieve it through permissions without adding more roles. I suggest we add a permission to show it to everyone/no-one and then when we add the manager role it'll be easy to configure this to only be shown to managers.

**Comment by sglangevin** [comment]
Right - we've been sort of hacking this for projects where we are using replication depth by adding a role to the managers that we then use to set replication depth only for them. Not sure if that will work in this case as well, but that's what I had in mind.

I agree - let's put in the permission for everyone/no one for now and then we can improve on that in the future when we add the manager role.

**Comment by dianabarsan** [comment]
Please review, @garethbowen . Thanks!

**Comment by dianabarsan** [comment]
I have a few questions about the indicated colors: some of them are very similar to other colors which are already used in the app. 

For example: 
 - unselected icons #A7A9AC - similar to #A5A8AA (used for deceased)
![image](https://user-images.githubusercontent.com/35681649/41082318-912996b0-6a36-11e8-9278-96d9721d6a49.png)
- verified with error color #E0381F - similar to #E33030 (used for high risk)
![image](https://user-images.githubusercontent.com/35681649/41082429-fd3858aa-6a36-11e8-9962-802cb373e391.png)
- verified correct icon color #8EB41D (indicated in the document) - similar to #8EB51C (kujua SMS green)
![image](https://user-images.githubusercontent.com/35681649/41082615-96bd690c-6a37-11e8-8618-438f1aaf5307.png)

@sglangevin , @amandacilek do you think we could reuse these already existent colors?

**Comment by amandacilek** [comment]
Thanks for asking. Aesthetically it makes no difference, as they are practically the same colors. Can we change all instances of the old existing colors to the new colors or is that tedious? That would be my preference.

I took my red and green hex codes from the "Foundation web colors" brand guide that Lester created:
![screen shot 2018-06-07 at 8 28 28 am](https://user-images.githubusercontent.com/26235957/41110069-5e61113a-6a2d-11e8-9676-4880d10955b9.png)

I then used those two colors in my new updated color guide for the app here (goes along with 3.1 nav update):
https://drive.google.com/open?id=15gBIUXiupi0Z7ectwNB5scLm6_1q7d_S

**Comment by sglangevin** [comment]
It might be easiest to stick with the existing colors for now and then we can update all of the colors to the new ones in 3.1. Otherwise the new colors will be partially implemented for a few versions.

**Comment by sglangevin** [comment]
Yeah, I didn’t realize that either!

**Comment by dianabarsan** [comment]
Merged into `2.16.x` and forward ported to `master`.

**Comment by ngaruko** [comment]
Feature seems to work with a glitch: it takes quite long for the choice (click on `Has errors` or `Correct`) takes a bit too long(about 5 secs) to show in the UI.

**Comment by garethbowen** [comment]
@ngaruko Were you logged in as an online or offline user at the time? If online then it's probably the server round-trip + replication time, and maybe we should add a loading spinner.

Possibly related: #3466

**Comment by ngaruko** [comment]
Online @garethbowen and true, it must be related to #3466. Will close this and test it again once #3466 has been fixed.

**Comment by sglangevin** [comment]
@ngaruko where was this tested? I see that beta-old.dev hasn't been updated to the 2.16.0 beta.
